### PR TITLE
Renovation: add ES2020 features using babel's plugins

### PR DESCRIPTION
**Pull request by LazyLahtak**
Add Nullish Coalescing Operator:
```js
const result = value ?? defaultValue
```
equal
```js
let result;
if (value !== null && value !== undefined) {
  result = value;
} else {
  result = defaultValue;
}
```
_________________________
Add Optional Chaining:
```js
const obj = {
  field: {
    value: 5,
  },
}

obj?.field?.value     // 5
obj?.subfield?.value  //undefined
```

```js
const obj = {
  func: () => { return 'someText'; },
}

obj?.func?.()         // 'someText'
obj?.anotherFunc?.()  // undefined
```

### CODE REVIEW

**Comment by LazyLahtak** [comment]
Ready to review

**Review by vconst — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -36,9 +36,14 @@
...
   },
   "devDependencies": {
+    "@babel/core": "^7.7.7",
+    "@babel/plugin-proposal-nullish-coalescing-operator": "^7.7.4",
+    "@babel/plugin-proposal-optional-chaining": "^7.7.5",
+    "@babel/preset-env": "^7.7.7",
     "@types/jquery": "^2.0.34",
     "angular": "^1.6.10",
-    "babel-core": "^6.26.0",
+    "babel-core": "^7.0.0-bridge.0",
```
Comment (San4es) [inline package.json:45]: Do we need both `@babel/core` and `babel-core` packages?

Comment (LazyLahtak) [reply package.json:45]: Thank you, fixed!

**Review by San4es — DISMISSED** [review]

**Comment by IlyaKhD** [comment]
LGTM
Btw, does `babel-eslint` take into account which babel plugins are installed?

**Review by IlyaKhD — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (LazyLahtak) [reply package.json:45]: UPD: we still need `babel-core` for `gulp`

**Comment by LazyLahtak** [comment]
> LGTM
> Btw, does `babel-eslint` take into account which babel plugins are installed?

Yes, it does.
We need to add all used plugins in the `babelrc`

**Review by churkin — APPROVED** [review]

**Review by San4es — APPROVED** [review]
Happy New "ES2020" Year!

**Pull request merged**